lib/bootvar.ml

```diff
@@ -34,12 +34,14 @@ let get_cmd_line () =
        Lwt.return cmdline)
 
 let create () = 
-  get_cmd_line () >>= fun cmd_line ->
-  let entries = Re_str.(split (regexp_string " ") cmd_line) in
+  get_cmd_line () >>= fun cmd_line_raw ->
+  (* Strip leading whitespace *)
+  let cmd_line = Astring.String.trim cmd_line_raw in
+  let entries = Astring.String.cuts ~empty:false ~sep:" " cmd_line in
   let parameters =
     List.map (fun x ->
-        match Re_str.(split (regexp_string "=") x) with 
-        | [a;b] -> (a,b)
+        match Astring.String.cut ~sep:"=" x with
+        | Some (a,b) -> (a,b)
```

Comment:
This will still not allow you to specify 1) variable values that have space in them 2) variable values that have `'='` in them. You need a real parser here, neither regexps nor string splits. You can adapt the second example [here](http://erratique.ch/software/astring/doc/Astring#examples) (`parse_env`).